ext/soap/php_encoding.c

```diff
@@ -274,6 +274,20 @@ static encodePtr find_encoder_by_type_name(sdlPtr sdl, const char *type)
 		    if (strcmp(enc->details.type_str, type) == 0) {
 				return enc;
 			}
+            int ns_len = strlen(enc->details.ns);
```

Comment:
I see three small issues here:
- For the lookup, we're allocating a new string for every item to compare against, and we're copying data. It's possible to do this without any allocations.
- Please don't use int to store a string length, this can overflow.
- The success case, i.e. when `enc` is returned, is not tested by your testcases.

I think for the encoder code we could use something like this:
```c
ZEND_HASH_FOREACH_PTR(sdl->encoders, enc)  {
	if (type[0] == '{') {
		size_t ns_len = strlen(enc->details.ns);
		if (strncmp(enc->details.ns, type + 1, ns_len) == 0
			&& type[ns_len + 1] == '}'
			&& strcmp(enc->details.type_str, type + ns_len + 2) == 0) {
			return enc;
		}
	} else {
		if (strcmp(enc->details.type_str, type) == 0) {
			return enc;
		}
	}
} ZEND_HASH_FOREACH_END();
```